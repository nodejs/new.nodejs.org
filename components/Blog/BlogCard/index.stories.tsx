import BlockQuote from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof BlockQuote>;
type Meta = MetaObj<typeof BlockQuote>;

export const Default: Story = {
  args: {
    author: 'Bat Man',
    category: 'category-mock',
    date: '2023-04-21 23:40:56.77',
    slug: '/blog/category-mock/sample-blog',
    title: 'Sample Test Blog',
    readingTime: '1 min read',
  },
};

export default { component: BlockQuote } as Meta;
