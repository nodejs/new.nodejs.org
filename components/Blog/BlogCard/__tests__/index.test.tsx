import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import BlogCard from '..';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({}),
}));

jest.mock('../../../../hooks/useLocale', () => ({
  useLocale: jest.fn().mockReturnValue({
    currentLocale: {},
  }),
}));

describe('BlogCard component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <IntlProvider locale="en" onError={() => {}}>
        <BlogCard
          author="Bat Man"
          category="category-mock"
          date="2023-04-21 23:40:56.77"
          slug="/blog/category-mock/sample-blog"
          title="Sample Test Blog"
          readingTime="1 min read"
        />
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
