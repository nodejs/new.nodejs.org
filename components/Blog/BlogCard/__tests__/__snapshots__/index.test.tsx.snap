// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard component renders correctly 1`] = `
<div>
  <div
    class="blogCard"
  >
    <div
      class="title"
    >
      <a
        href="/undefined/blog/category-mock/sample-blog"
      >
        Sample Test Blog
      </a>
      <div
        class="metadata"
      >
        <a
          class="category"
          href="/undefined/blog/category-mock"
        >
          category-mock
        </a>
        <span>
          1 min read
        </span>
      </div>
    </div>
    <div
      class="content"
    >
      <h4>
        April 21, 2023
      </h4>
      <p>
        components.blog.blogCard.author.by
         
        <span>
          Bat Man
        </span>
      </p>
    </div>
  </div>
</div>
`;
